tools/harness-automation/autothreadharness/runner.py

```diff
@@ -106,10 +106,17 @@ def add_result(self, test, passed, error=None):
             test (TestCase): The test just run
             passed (bool): Whether the case is passed
         """
+        if str(test.__class__.__name__) in self.result and 'fails' in self.result[str(test.__class__.__name__)]:
+            fails = int(self.result[str(test.__class__.__name__)]['fails'])
+        else:
+            fails = 0
+        if passed is False:
```

Comment:
nit: it's better just use `if not passed:` according to [PEP8](https://www.python.org/dev/peps/pep-0008/)
```
Don't compare boolean values to True or False using ==.

Yes:   if greeting:
No:    if greeting == True:
Worse: if greeting is True:
```

Reply:
The reason I used "is False" is that "addError" will pass "None" to "passed" .  "passed" has 3 kinds of value : True, False and None. Only if passed is False, fails ++